diff --git a/src/components/ui/button.tsx b/src/components/ui/button.tsx
index 1991486d27f1bb3232f20bc6ea16b13e2e0471f7..a79fd7cf91f2ca974837c5e5e6db443e7699e325 100644
--- a/src/components/ui/button.tsx
+++ b/src/components/ui/button.tsx
@@ -46,15 +46,13 @@ function Button({
   VariantProps<typeof buttonVariants> & {
     asChild?: boolean;
   }) {
-  const Comp = asChild ? Slot : 'button';
+  const classNames = cn(buttonVariants({ variant, size, className }));
 
-  return (
-    <Comp
-      data-slot="button"
-      className={cn(buttonVariants({ variant, size, className }))}
-      {...props}
-    />
-  );
+  if (asChild) {
+    return <Slot data-slot="button" className={classNames} {...props} />;
+  }
+
+  return <button data-slot="button" className={classNames} {...props} />;
 }
 
 type ButtonProps = React.ComponentProps<typeof Button>;
